balancer/catabalancer/catalyst_balancer.go

```diff
@@ -421,58 +421,76 @@ func StartMetricSending(nodeName string, latitude float64, longitude float64, mi
 	ticker := time.NewTicker(updateNodeStatsEvery)
 	go func() {
 		for range ticker.C {
-			start := time.Now()
-			sysusage, err := GetSystemUsage()
-			if err != nil {
-				log.LogNoRequestID("catabalancer failed to get sys usage", "err", err)
+			done := make(chan bool)
```

Comment:
I think the more "golang way" of doing the same would be passing context with timeout into `sendMetrics()`. So you could have something like this:

```golang
ctx, cancel := context.WithTimeout(context.Background(), timeout)
sendMetrics(ctx, nodeName, latitude, longitude, mist, nodeStatsDB)
```

And then inside `sendMetrics()` to have

```golang
GetSystemUsage(ctx)
```

And so on. The good thing about this approach would be that in case of the timeout you cancel all system and network calls.

Reply:
@leszko I've pretty much re-done this PR, please could you take a look? thanks